### ISSUE TITLE
SC2096: warns for '-e -x' but not '-ex'

### ISSUE THREAD

**Issue by austin987** [koalaman/shellcheck#805]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash -ex
echo a
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
No issues detected!
```

#### Here's what I wanted or expected to see:
```sh
In foo2.sh line 1:
#!/bin/bash -ex
^-- SC2096: On most OS, shebangs can only specify a single parameter.
```

This currently does warn if the options are provided separately (-e -x), but not if passed together as one argument (-ex).

**Comment by andlrc** [comment]
>This currently does warn if the options are provided separately (-e -x), but not if passed together as one argument (-ex).

That is because `-ex` is a single argument?

**Comment by austin987** [comment]
It's passed as one argument, GNU style, but it doesn't work either:
```sh
austin@austin2:~$ bash foo.sh 
a
b

austin@austin2:~$ ./foo.sh 
+ echo a
a
+ false

austin@austin2:~$ cat foo.sh 
#!/bin/bash -ex
echo a
false
echo b
```

but putting it inline works:
```sh
austin@austin2:~$ ./foo2.sh 
+ set -e
+ echo a
a
+ false

austin@austin2:~$ bash foo2.sh 
+ set -e
+ echo a
a
+ false

austin@austin2:~$ cat foo2.sh 
#!/bin/bash
set -x
set -e

echo a
false
echo b
```

and as one option, inline:
```sh
austin@austin2:~$ ./foo3.sh 
+ echo a
a
+ false

austin@austin2:~$ bash foo3.sh 
+ echo a
a
+ false

austin@austin2:~$ cat foo3.sh 
#!/bin/bash
set -xe

echo a
false
echo b
```

**Comment by koalaman** [comment]
`bash filename` will always ignore the shebang entirely. There's no way around that.

**Comment by austin987** [comment]
Good to know, thanks.
